collectors/ebpf.plugin/ebpf.c

```diff
@@ -1092,12 +563,147 @@ static void parse_args(int argc, char **argv)
     }
 }
 
+/**
+ * Fill the ebpf_functions structure with default values
+ *
+ * @param ef the pointer to set default values
+ */
+void fill_ebpf_functions(ebpf_functions_t *ef) {
+    memset(ef, 0, sizeof(ebpf_functions_t));
+    ef->kernel_string = kernel_string;
+    ef->running_on_kernel = running_on_kernel;
+    ef->map_fd = callocz(EBPF_MAX_MAPS, sizeof(int));
+    ef->isrh = isrh;
+}
+
+/**
+ * Define how to load the ebpf programs
+ *
+ * @param ptr the option given by users
+ */
+static inline void how_to_load(char *ptr) {
+    if (!strcasecmp(ptr, "return"))
+        ebpf_set_thread_mode(MODE_RETURN);
+}
+
+/**
+ * Set collector values
+ */
+static void set_collector_values() {
+    struct section *sec = collector_config.first_section;
+    int disable_apps = 0;
+    int enabled = 0;
+    while(sec) {
+        struct config_option *values;
+        if(!strcasecmp(sec->name, "global")) {
+            values = sec->values;
+            while(values) {
+                if (!strcasecmp(values->name, "load"))
```

Comment:
should we rename the parameter? just having `load` in config is not self explanatory. Maybe `ebpf load mode`?

Reply:
We can, but we need to keep backward compatibility.

Reply:
how about reading `load` for bw compat and introduce also `ebpf load mode`?

Reply:
Yes, we will do this.

Reply:
> but we need to keep backward compatibility.

do we really need? ebpf is in alpha, i think it is ok to drop some compatibility. My point that if there a chance to not be _backward compatible_ i would suggest not to be - to keep code cleaner.

If there is no way - i suggest to not mention old `load` in the docs.

Reply:
I'm pretty sure it's not in alpha anymore though.

Reply:
I liked your suggestion @ilyam8 and I updated the documentation.

Reply:
Guys, let us keep both, it is only one line, but the documentation will have reference to @underhood  suggestion only.